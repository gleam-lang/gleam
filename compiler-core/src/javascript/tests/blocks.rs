use crate::assert_js;

#[test]
fn block() {
    assert_js!(
        r#"
fn go() {
  let x = {
    1
    2
  }
  x
}
"#,
        r#""use strict";

function go() {
  let x = (() => {
    1;
    return 2;
  })();
  return x;
}
"#
    );
}

#[test]
fn sequences() {
    assert_js!(
        r#"
fn go() {
  "one"
  "two"
  "three"
}
"#,
        r#""use strict";

function go() {
  "one";
  "two";
  return "three";
}
"#
    );
}

#[test]
fn left_operator_sequence() {
    assert_js!(
        r#"
fn go() {
  1 == {
    1
    2
  }
}
"#,
        r#""use strict";

function go() {
  return 1 === (() => {
    1;
    return 2;
  })();
}
"#
    );
}

#[test]
fn right_operator_sequence() {
    assert_js!(
        r#"
fn go() {
  {
    1
    2
  } == 1
}
"#,
        r#""use strict";

function go() {
  return (() => {
    1;
    return 2;
  })() === 1;
}
"#
    );
}
