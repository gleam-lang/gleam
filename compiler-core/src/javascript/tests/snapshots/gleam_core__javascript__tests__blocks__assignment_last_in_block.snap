---
source: compiler-core/src/javascript/tests/blocks.rs
expression: "\npub fn main() {\n  let a = {\n    let b = 1\n    let c = b + 1\n  }\n  a\n}\n"
---
----- SOURCE CODE

pub fn main() {
  let a = {
    let b = 1
    let c = b + 1
  }
  a
}


----- COMPILED JAVASCRIPT
export function main() {
  let _block;
  {
    let b = 1;
    let c = b + 1;
    _block = c;
  }
  let a = _block;
  return a;
}
